Accordion.js

```diff
@@ -0,0 +1,474 @@
+			panel.containerNode.setAttribute("aria-hidden", "true");
+			return panel;
+		},
+
+		_numNonUpgradedChildren: 0,
+
+		_setupNonUpgradedChild: function (panel) {
+			panel.accordion = this;
+			this._numNonUpgradedChildren++;
+			panel.addEventListener("customelement-attached", this._attachedlistener = function () {
+				this.removeEventListener("customelement-attached", this._attachedlistener);
+				this.accordion._panelList.push(this.accordion._setupUpgradedChild(this));
+				this.accordion._numNonUpgradedChildren--;
+			}.bind(panel));
+		},
+
+		attachedCallback: function () {
+			this._panelList = [];
+			// Declarative case (panels specified declaratively inside the declarative Accordion)
+			var panels = this.querySelectorAll("d-panel");
+			if (panels) {
+				for (var i = 0, l = panels.length; i < l; i++) {
+					if (!panels[i].attached) {
+						this._setupNonUpgradedChild(panels[i]);
+					} else {
+						this._panelList.push(this._setupUpgradedChild(panels[i]));
+					}
+				}
+			}
+		},
+
+		_showOpenPanel: function () {
+			//The default open panel is the first one
+			if (!this._selectedChild && this._panelList.length > 0) {
+				this._selectedChild = this._panelList[0];
+			}
+			//Show selectedChild if exists
+			if (this._selectedChild && this._selectedChild.attached) {
+				this.show(this._selectedChild);
+			}
+		},
+
+		/* jshint maxcomplexity: 13 */
+		refreshRendering: function (props) {
+			if ("selectedChildId" in props) {
+				var childNode = this.ownerDocument.getElementById(this.selectedChildId);
+				if (childNode) {
+					if (childNode.attached) {
+						if (childNode !== this._selectedChild) { //To avoid calling show() method twice
+							if (!this._selectedChild) { //If selectedChild is not initialized, then initialize it
+								this._selectedChild = childNode;
+							}
+							this.show(childNode);
+						}
+					} else {
+						this._selectedChild = childNode;
+					}
+				}
+			}
+			if ("attached" in props || "_numNonUpgradedChildren" in  props) {
+				if (this._numNonUpgradedChildren === 0) {
+					this._showOpenPanel();
+				}
+			}
+			if ("openIconClass" in props) {
+				this.getChildren().forEach(function (panel) {
+					if (panel.attached && !panel.iconClass) {
+						panel.headerNode.checkedIconClass = this.openIconClass;
+					}
+				}.bind(this));
+			}
+			if ("closedIconClass" in props) {
+				this.getChildren().forEach(function (panel) {
+					if (panel.attached && !panel.closedIconClass) {
+						panel.headerNode.iconClass = this.closedIconClass;
+					}
+				}.bind(this));
+			}
+			if ("singleOpen" in props) {
+				this.setAttribute("aria-multiselectable", !this.singleOpen);
+				if (this.singleOpen) {
+					this._showOpenPanel();
+					this._panelList.forEach(function (panel) {
+						if (panel.open && panel !== this._selectedChild) {
+							this.hide(panel);
+						}
+					}.bind(this));
+				}
+			}
+		},
+		/* jshint maxcomplexity: 10 */
+
+		_supportAnimation: function () {
```

Comment:
Maybe we could change the name of the method to `_useAnimation()` and move into it also the validation of `this.animate` ?